packages/eip-7702-internal-rpc-middleware/src/hooks/upgradeAccount.test.ts

```diff
@@ -0,0 +1,370 @@
+import { rpcErrors, providerErrors } from '@metamask/rpc-errors';
+import type {
+  JsonRpcRequest,
+  PendingJsonRpcResponse,
+  Json,
+} from '@metamask/utils';
+
+import { upgradeAccount } from './upgradeAccount';
+
+const TEST_ACCOUNT = '0x1234567890123456789012345678901234567890';
+const UPGRADE_CONTRACT = '0x0000000000000000000000000000000000000000';
+
+const createUpgradeHandler = () => {
+  const end = jest.fn();
+  const next = jest.fn();
+  const upgradeAccountFn = jest.fn();
+  const getCurrentChainIdForDomain = jest.fn().mockReturnValue('0x1');
+  const getAccounts = jest.fn().mockResolvedValue([TEST_ACCOUNT]);
+  const isEip7702Supported = jest
+    .fn()
+    .mockImplementation(async ({ chainIds }: { chainIds: string[] }) => {
+      return chainIds.map((chainId: string) => {
+        if (chainId === '0x1' || chainId === '0xaa36a7') {
+          return {
+            chainId,
+            isSupported: true,
+            upgradeContractAddress: UPGRADE_CONTRACT,
+          };
+        }
+        return {
+          chainId,
+          isSupported: false,
+        };
+      });
+    });
+
+  const response = {
+    result: null,
+    id: 1,
+    jsonrpc: '2.0' as const,
+  } as PendingJsonRpcResponse;
+  const handler = async (
+    request: JsonRpcRequest<Json[]> & { origin: string },
+  ) => {
+    try {
+      const result = await upgradeAccount(request, response, {
+        upgradeAccount: upgradeAccountFn,
+        getCurrentChainIdForDomain,
+        isEip7702Supported,
+        getAccounts,
+      });
+      response.result = result;
+    } catch (error) {
+      end(error);
+    }
+  };
+
+  return {
+    end,
+    next,
+    upgradeAccount: upgradeAccountFn,
+    getCurrentChainIdForDomain,
+    isEip7702Supported,
+    getAccounts,
+    response,
+    handler,
+  };
+};
+
+describe('upgradeAccount', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
```

Comment:
Jest is already configured to clear and restore mocks in this repo, we don't need to do this manually.

``` suggestion
```